lib/network_monitor.c

```diff
@@ -525,16 +528,15 @@ static void mptcpd_interface_callback(void *data, void *user_data)
  *
  * @return @c true if network interface is ready, and @c false other.
  */
-static bool is_interface_ready(struct ifinfomsg const *ifi)
+static bool is_interface_ready(struct mptcpd_nm const *nm,
+                               struct ifinfomsg const *ifi)
 {
-        /*
-          Only accept non-loopback network interfaces that are
-          up and running.
-        */
-        static unsigned int const iff_ready = IFF_UP | IFF_RUNNING;
+        // Only accept network interfaces that are up and running.
+        static unsigned int iff_ready = IFF_UP | IFF_RUNNING;
 
         return (ifi->ifi_flags & iff_ready) == iff_ready
-                && (ifi->ifi_flags & IFF_LOOPBACK) == 0;
+                && (nm->monitor_loopback
+                    || (ifi->ifi_flags & IFF_LOOPBACK) == 0);
```

Comment:
From a readability perspective, I kind of expect to see the `ifi_flags` checked first as the common case (and also that it's a tiny bit closer to the existing code). That it's also a micro-optimization at runtime is not too relevant as this is not a hot code path :)

If you have a rationale for the ordering here, it's fine to leave as-is.

Reply:
I have no particular rationale for the ordering.  Better to optimize for the common case, per your comment.